### ISSUE TITLE
Go client: wrong type definition for Lists

### ISSUE THREAD

**Issue by averzicco** [GoogleCloudPlatform/k8s-config-connector#422]
## Checklist

- [x] I did not find a related open issue.
- [x] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [ ] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

## Bug Description

I'm trying to use the go client package but the type definition for Lists of resources looks wrong. For example for [computeInstanceList](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f158208856e727336001b17397cbaa4c918d2ba4/pkg/apis/compute/v1beta1/computeinstance_types.go#L274-L278) you have
```
type ComputeInstanceList struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`
	Items             []ComputeInstance `json:"items"`
}
```

I would expect `metav1.ListMeta` instead of `metav1.ObjectMeta` for Lists

### Config Connector Version

GO client version: v1.41.0

**Comment by xiaobaitusi** [comment]
Hi @averzicco, thanks for reporting the issue. We will look into the fix for it and update this thread when we have more information. Thanks

**Comment by aa250816** [comment]
Yeah this bug prevents us from using the go types as we can't list the objects using the go client. This prevents us from setting watches.

**Comment by guilhem** [comment]
For the record, this bug froze my controller with this error:
```
pkg/mod/k8s.io/client-go@v0.20.5/tools/cache/reflector.go:167: Failed to watch *v1beta1.SQLDatabase: failed to list *v1beta1.SQLDatabase: returned object must be a list: object does not implement the List interfaces
```

**Comment by caieo** [comment]
@aa250816, @guilhem, and @averzicco, we're really sorry this bug was introduced with our go-clients. We have a fix staged for the next release and will update this thread when the release is out. Thank you for your patience!

**Comment by averzicco** [comment]
This bug has been fixed with [v1.45.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.45.0) release
